src/Processors/Formats/Impl/MsgPackRowInputFormat.cpp

```diff
@@ -496,10 +496,12 @@ DataTypePtr MsgPackSchemaReader::getDataType(const msgpack::object & object)
         case msgpack::type::object_type::EXT:
         {
             msgpack::object_ext object_ext = object.via.ext;
-            if (object_ext.type() == int8_t(MsgPackExtensionTypes::UUID))
+            if (object_ext.type() == int8_t(MsgPackExtensionTypes::UUIDType))
                 return std::make_shared<DataTypeUUID>();
             throw Exception(ErrorCodes::BAD_ARGUMENTS, "Msgpack extension type {%x} is not supported", object_ext.type());
         }
+        default:
```

Comment:
This do not compile in our CI, please double check if this is necessary, maybe we can use `__builtin_unreachable()` here.

Reply:
I think it is because `default` overides `case mysgpack::type::object_type::NIL`
Maybe we can remove `case mysgpack::type::object_type::NIL`

Reply:
No we should not do that, because that way if we add another value in enum we do not get warning, lets handle `case mysgpack::type::object_type::NIL` as before, remove default, and put __builtin_unreachable if necessary.